session/session.go

```diff
@@ -127,16 +127,6 @@ type Session interface {
 	ExecuteStmt(context.Context, ast.StmtNode) (sqlexec.RecordSet, error)
 	// Parse is deprecated, use ParseWithParams() instead.
 	Parse(ctx context.Context, sql string) ([]ast.StmtNode, error)
-	// ParseWithParams is the parameterized version of Parse: it will try to prevent injection under utf8mb4.
-	// It works like printf() in c, there are following format specifiers:
-	// 1. %?: automatic conversion by the type of arguments. E.g. []string -> ('s1','s2'..)
-	// 2. %%: output %
-	// 3. %n: for identifiers, for example ("use %n", db)
-	//
-	// Attention: it does not prevent you from doing parse("select '%?", ";SQL injection!;") => "select '';SQL injection!;'".
-	// One argument should be a standalone entity. It should not "concat" with other placeholders and characters.
-	// This function only saves you from processing potentially unsafe parameters.
-	ParseWithParams(ctx context.Context, sql string, args ...interface{}) ([]ast.StmtNode, error)
```

Comment:
What is the reason to remove it in the interface?

Reply:
The external SQL all use `ExecuteStmt` or `ExecutePrepared` ...

And we have the `ExecuteInternal` for internal SQL already, and also the `RestrictedSQLExecutor`, there is no need to include this method in the interface.

Reply:
Do we know that we can convert most uses of `Execute` to `ExecuteInternal` safely? My understanding is that `ExecuteInternal` is similar to Restricted, which is that it does not binary log, runs in the server permissions etc.

If we expect that `COM_QUERY` is the only valid case of "Execute", then that's fine. It is already switched over to `ExecuteStmt`.

Reply:
OK, but you still leave it in the `RestrictedSQLExecutor` interface. I would like both-existing or both-none.

And you can completely extract SQL execution methods into another subinterface... Session could embed all sql executor interfaces,

Reply:
We can clean up the code after the refactor finish:

Step1: introduce the new API for `RestrictedSQLExecutor`
Step2: use the new API to replace the old one
Step3: remove the old API

We can embed `RestrictedSQLExecutor` into the `Session` interface like this:

```
type Session  interface {
    RestrictedSQLExecutor
    ...
}
```

Keep each interface small and composable.... after we clean the usage of the old ones.

Reply:
Sweet!